src/Schema/Collections/OptionallyUnqualifiedNamedObjectSet.php

```diff
@@ -110,6 +112,12 @@ public function toList(): array
         return $this->elements;
     }
 
+    /** {@inheritDoc} */
+    public function getIterator(): Traversable
+    {
+        return new ArrayIterator($this->elements);
```

Comment:
Using a generator might be a bit more lightweight.

```suggestion
        yield from $this->elements;
```